packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -1099,7 +1099,36 @@ export function simplify<T>(
 			}
 			return { kind: 'union', members: members }
 		case 'struct':
-			const fields: SimplifiedStructTypePairField[] = []
+			const literalFields = new Map<string, StructTypePairField>()
+			let complexFields: SimplifiedStructTypePairField[] = []
+
+			function addField(
+				key: string | SimplifiedMcdocType,
+				field: StructTypePairField,
+			) {
+				if (typeof key === 'string') {
+					literalFields.set(key, field)
+				} else if (key.kind === 'literal' && key.value.kind === 'string') {
+					literalFields.set(key.value.value, field)
+				} else if (key.kind === 'union') {
+					key.members.forEach(m => addField(m, field))
+				} else {
+					complexFields = complexFields.filter(other =>
```

Comment:
Not sure if intentional, but this will keep fields whose keys are assignable to previous types if the previous type is broader.

For example, this would not be deduplicated:
```rs
struct {
    [string]: int
    [#[id=block] string]: int
}
```

This is because you are only checking if `string` is assignable to `#[id=block] string` (which it won't be when attributes are implemented), but not the other way around.

I think if we want to allow for this, it would make more sense to check the other way around and allow this:

```rs
struct {
    [string]: int
    [#[id=block] string]: int
}
```

Reply:
You're right, that way around makes more sense 👍